src/dbridge_js/autogram/autogram-implementation.ts

```diff
@@ -92,35 +87,36 @@ export class DBridgeAutogramImpl implements ImplementationInterface {
 
   public addObject(obj: InputObject, callback: OnSuccessCallback): void {
     if (this.signRequest.signingStatus == SigningStatus.signed) {
-      console.warn("Resetting sign request");
+      log.warn("Resetting sign request");
       this.resetSignRequest();
     }
 
     if (this.signRequest.signingStatus !== SigningStatus.new) {
-      console.error("Adding object to non-new sign request");
+      log.error("Adding object to non-new sign request");
     }
-    console.log(obj);
+    log.info(obj);
     this.signRequest.addObject(obj);
-    console.log(callback);
+    log.debug(callback);
     callback.onSuccess();
   }
 
   public async getSignature(
-    parameters: PartialSignerParameters,
+    parameters: Partial<DesktopSignatureParameters>,
     callback: OnSuccessCallback1,
     decodeBase64 = false
   ): Promise<void> {
-    const signingMethod = await this.ui.startSigning();
-    if (signingMethod === SigningMethod.reader) {
-      const abortController = new AbortController();
-      this.ui.desktopSigning(abortController);
-      await this.launchDesktop(abortController);
-      return this.getSignatureDesktop(parameters, callback, decodeBase64);
-    } else if (signingMethod === SigningMethod.mobile) {
-      return this.getSignatureMobile(parameters, callback, decodeBase64);
-    } else {
-      console.log("Invalid signing method");
-      throw new Error("Invalid signing method");
+    try {
+      const response = await this.client.sign(
+        this.signRequest.document,
+        this.signRequest.signatureParameters(parameters),
+        this.signRequest.payloadMimeType,
+        decodeBase64
+      );
+      this.signedObject = response;
```

Comment:
this is part of the reason why stateful API is maybe not so good idea - now we handle SignRequest in both app and SDK